File: POO/Py/Persona.py
class Persona:
    def __init__(self, nombre, edad, carrera, universidad):
        self.nombre = nombre
        self.edad = edad
        self.carrera = carrera
        self.universidad = universidad

    def saludar(self):
        return f"Hola, soy {self.nombre}"

    def cumpleanios(self):
        return f"Yo tengo {self.edad} años."

    def estudiar(self):
        return f"Estoy estudiando {self.carrera}"

    def mostrarUniversidad(self):
        return f"Estoy estudiando en la {self.universidad}"

    def modificarNombre(self):
        while True:
            nuevo_nombre = input("Ingresa tu nombre: ")
            if self.validarTexto(nuevo_nombre):
                self.nombre = nuevo_nombre
                break
            else:
                print("Caracter no valido, ingrese solamente letras.")
        return self.saludar()

    def modificarEdad(self):
        while True:
            try:
                nueva_edad = int(input("Ingresa tu edad: "))
                if 0 <= nueva_edad <= 118:
                    self.edad = nueva_edad
                    break
                else:
                    print("Edad no valida, ingrese un número entre 0 y 118.")
            except ValueError:
                print("Por favor ingrese un número válido.")
        return f"¡Felicidades! Tienes {self.edad} años."

    def modificarCarrera(self):
        while True:
            nueva_carrera = input("Ingresa tu carrera: ")
            if self.validarTexto(nueva_carrera):
                self.carrera = nueva_carrera
                break
            else:
                print("Caracter no valido, ingrese solamente letras.")
        return self.estudiar()

    def modificarUniversidad(self):
        nueva_universidad = input("Ingresa tu universidad: ")
        self.universidad = nueva_universidad
        return f"Estas estudiando en la {self.universidad}"

    def validarTexto(self, texto):
        return texto.replace(" ", "").isalpha()

    def eliminarNombre(self):
        self.nombre = None

    def eliminarEdad(self):
        self.edad = None

    def eliminarCarrera(self):
        self.carrera = None

    def eliminarUniversidad(self):
        self.universidad = None

# Crear una instancia de la clase Persona
persona = Persona('Marco', 19, 'Ingenieria industrial', 'Universidad Catolica Boliviana')

# Ejemplos de uso de los métodos
print(persona.saludar())
print(persona.modificarNombre())
print(persona.modificarEdad())
print(persona.modificarCarrera())
print(persona.modificarUniversidad())

File: POO/JS/Persona.js
class Persona {
  constructor(nombre, edad, carrera, universidad) {
    this.nombre = nombre;
    this.edad = edad;
    this.carrera = carrera;
    this.universidad = universidad;
  }

  // Metodo avanzado en clases
  saludar() {
    return 'Hola, soy ' + this.nombre + ', y tengo ' + this.edad + ' años.';
  }

  cumpleanios() {
    return 'Yo tengo ' + this.edad + ' años.';
  }

  estudiar() {
    return 'Estoy estudiando ' + this.carrera;
  }

  mostrarUniversidad() {
    return 'Estoy estudiando en la ' + this.universidad;
  }

  // Metodos para modificar los atributos
  ModificarNombre() {
    const nuevoNombre = prompt("Ingresa tu nombre");
    if (this.validarTexto(nuevoNombre)) {
      this.nombre = nuevoNombre;
    } else {
      alert("Caracter no valido, ingrese solamente letras");
    }
    return this.saludar();
  }

  ModificarEdad(){
    const nuevaEdad = Number(prompt("Ingresa tu edad"));
    if (nuevaEdad >= 0) {
      this.edad = nuevaEdad;
    } else {
      alert("Edad no valida, ingrese un numero positivo");
    }
    return `¡Felicidades! Tienes ${this.edad} años.`;
  }

  ModificarCarrera() {
    const nuevaCarrera = prompt("Ingresa tu carrera");
    if (this.validarTexto(nuevaCarrera)) {
      this.carrera = nuevaCarrera;
    } else {
      alert("Caracter no valido, ingrese solamente letras");
    }
    return this.estudiar();
  }

  ModificarUniversidad(){
    this.universidad = prompt("Ingresa tu universidad");
    return 'Estas estudiando en la ' + this.universidad;
  }

  // Metodo para validar texto
  validarTexto(texto) {
    return /^[a-zA-Z\s]+$/.test(texto);
  }

  // Metodos para eliminar los atributos
  EliminarNombre() {
    this.nombre = undefined;
  }

  EliminarEdad(){
    this.edad = undefined;
  }

  EliminarCarrera(){
    this.carrera = undefined;
  }

  EliminarUniversidad(){
    this.universidad = undefined;
  }
}

// Crear una instancia de la clase Persona
const persona = new Persona('Marco', 19, 'Ingenieria industrial', 'Universidad Catolica Boliviana');

// Inicializar los elementos HTML
const saludar = document.getElementById('saludar');
const edad = document.getElementById('edad');
const carrera = document.getElementById('carrera');
const universidad = document.getElementById('universidad');

// Funcion para saludar
function botonSaludar() {
  saludar.textContent = persona.saludar();
}

// Funcion para cumpleaños
function botonCumpleanios() {
  edad.textContent = persona.cumpleanios();
}

// Funcion para estudiar
function botonEstudiar() {
  carrera.textContent = persona.estudiar();
}

// Funcion para mostrar universidad
function botonUniversidad() {
  universidad.textContent = persona.mostrarUniversidad();
}

// Funcion para modificar nombre
function botonModificarNombre(){
  saludar.textContent = persona.ModificarNombre();
}

// Funcion para modificar edad
function botonModificarEdad(){
  edad.textContent = persona.ModificarEdad();
}

// Funcion para modificar carrera
function botonModificarCarrera(){
  carrera.textContent = persona.ModificarCarrera();
}

// Funcion para modificar universidad
function botonModificarUniversidad(){
  universidad.textContent = persona.ModificarUniversidad();
}

// Funcion para eliminar nombre
function botonEliminarNombre(){
  saludar.textContent = ''; 
}

// Funcion para eliminar edad
function botonEliminarEdad(){
  edad.textContent = ''; 
}

// Funcion para eliminar carrera
function botonEliminarCarrera(){
  carrera.textContent = ''; 
}

// Funcion para eliminar universidad
function botonEliminarUniversidad(){
  universidad.textContent = ''; 
}